components/WithdrawLiquidity/WithdrawLiquidityController.tsx

```diff
@@ -1,24 +1,183 @@
+
+const ZEROMIN = 0;
 
 const WithdrawLiquidity = () => {
-  const [sliderPercentage, setsliderPercentage] = useState<number>(0)
+  const [depPool, setdepPool] = useState<Contract | null>(null)
+  const [sliderPercentage, setsliderPercentage] = useState([0])
+  const { provider, accountInfo } = useContext(WalletContext)
+  const [uniPrice, setUniPrice] = useState<string>("0")
+  const [liquidityAmt, setLiquidityAmt] = useState("0");
+  const [liqInTokB, setLiqInTokB] = useState("0");
 
   async function changeSliderPercentage(percentage: number) {
     let data: number = parseInt(percentage.toFixed(0))
-    setsliderPercentage(data)
+    setsliderPercentage([data])
+  }
+
+  function sliderPercentChange(values: number[]) {
+    setsliderPercentage(values)
+  }
+
+  async function withdrawLiquidity(balance: number) {
+    if (provider) {
+      let address = "0x3eFadc5E507bbdA54dDb4C290cc3058DA8163152"
+      if (accountInfo  && accountInfo?.address) {
+        console.log(balance + " My")
+          setdepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo?.address)))
+      } else {
+        console.log(balance)
+          setdepPool(new ethers.Contract(address, DepPool.abi, provider))
+      }
+  } else {
+      console.log("Please connect wallet")
+  }
+
+    if (!accountInfo || !accountInfo.address) {
+      console.log("Wallet not connected.")
+      return
+  }
+
+    if (depPool === null) {
+      return
+    }
+    await approveWithdraw(depPool, depPool._address)
+
+    await depPool
+    .removeLiquidity(
+      ethers.utils.parseEther(balance.toString()),
+      ZEROMIN,
+      ZEROMIN,
+      accountInfo.address
+    )
+    .send({ from: accountInfo.address })
+    .then((res: any) => {
+        return res
+    })
+    .catch((err: any) => {
+        console.error(err)
+    })
+  }
+
+  async function approveWithdraw(depPool: Contract | null, depPoolAddr: string) {
+    if (!accountInfo || !accountInfo.address) {
+      console.log("Wallet not connected.")
+      return
+  }
+    if (depPool === null) {
+      return null;
+    } else {
+      if (provider !== null) {
+        let tokA = "0xbed4729d8E0869f724Baab6bA045EB67d72eCb7c"
+        approve(tokA, depPoolAddr);
+  } else {
+    console.log("Please connect wallet")
+  }
+    }
   }
 
-  async function onDone(percent: number) {
-    console.log(`I'm done. here's the percent: ${percent}%`);
-  };
+  async function approve(fromToken: string, toAddr: string) {
+    if (!provider) {
+        console.log("provider or accountInfo not set")
+        return
+    }
+    if (!accountInfo || !accountInfo.address) {
+      console.log("Wallet not connected.")
+      return
+  }
+    console.log(fromToken)
+    var erc20 = new ethers.Contract(fromToken, IERC20.abi, provider.getSigner(accountInfo?.address))
+    const res = await erc20.approve(toAddr, constants.MaxUint256).send({ from: accountInfo.address })
+    console.log("approved", res)
+}
+
+  useEffect(() => {
```

Comment:
Personal preference: generally I like putting useEffects at the top after the consts.